packages/ndk/lib/domain_layer/entities/filter.dart

```diff
@@ -73,6 +79,23 @@ class Filter {
     since = map['since'];
     until = map['until'];
     limit = map['limit'];
+
+    // Handle arbitrary tags
+    arbitraryTags = {};
+    map.forEach((key, value) {
+      if (key.startsWith('#') &&
```

Comment:
Dont restrict which tags are allowed, only check that it `startsWith` `#` and `length == 2`